src/query/execute_schema.c

```diff
@@ -10016,6 +10021,203 @@ do_alter_change_tbl_comment (PARSER_CONTEXT * const parser, PT_NODE * const alte
+      error = er_errid ();
+      goto exit;
+    }
+
+  error = locator_flush_class (class_obj);
+  if (error != NO_ERROR)
+    {
+      /* don't overwrite error */
+      goto exit;
+    }
+
+  /* force exclusive lock on class, even though it should have been already acquired */
+  if (locator_fetch_class (class_obj, DB_FETCH_WRITE) == NULL)
+    {
+      error = ER_FAILED;
+      goto exit;
+    }
+
+  error = tran_system_savepoint (UNIQUE_SAVEPOINT_CHANGE_COLUMN_COMMENT);
+  if (error != NO_ERROR)
+    {
+      goto exit;
+    }
+  tran_saved = true;
+
+  attr_node = alter_node->info.alter.alter_clause.attr_mthd.attr_def_list;
+  while (attr_node != NULL)
+    {
+      ctemplate = dbt_edit_class (class_obj);
+      if (ctemplate == NULL)
+	{
+	  /* when dbt_edit_class fails (e.g. because the server unilaterally aborts us), we must record the associated
+	   * error message into the parser. Otherwise, we may get a confusing error msg of the form: "so_and_so is not a
+	   * class". */
+	  pt_record_error (parser, parser->statement_number - 1, alter_node->line_number, alter_node->column_number,
+			   er_msg (), NULL);
+	  error = er_errid ();
+	  goto exit;
+	}
+
+      attr_name = get_attr_name (attr_node);
+
+      comment_node = attr_node->info.attr_def.comment;
+
+      assert (comment_node != NULL && comment_node->node_type == PT_VALUE);
+
+      /* get the attribute structure */
+      error =
+	smt_find_attribute (ctemplate, attr_name, (attr_chg_prop.name_space == ID_CLASS_ATTRIBUTE) ? 1 : 0,
+			    &found_attr);
+      if (error != NO_ERROR)
+	{
+	  return error;
+	}
+
+      assert (found_attr != NULL);
+
+      attr_chg_prop.name_space = found_attr->header.name_space;
+
+      if (attr_node->info.attr_def.attr_type == PT_NORMAL)
+	{
+	  attr_chg_prop.new_name_space = ID_ATTRIBUTE;
+	}
+      else if (attr_node->info.attr_def.attr_type == PT_SHARED)
+	{
+	  attr_chg_prop.new_name_space = ID_SHARED_ATTRIBUTE;
+	}
+
+      /* consolidate properties : */
+      {
+	int i = 0;
+
+	for (i = 0; i < NUM_ATT_CHG_PROP; i++)
+	  {
+	    int *const p = &(attr_chg_prop.p[i]);
+
+	    *p = 0;
+	    *p |= ATT_CHG_PROPERTY_PRESENT_OLD;
+	    *p |= ATT_CHG_PROPERTY_UNCHANGED;
+	  }
+
+	attr_chg_prop.p[P_COMMENT] &= ~ATT_CHG_PROPERTY_UNCHANGED;
+      }
+
+      /* comment */
+      attr_chg_prop.p[P_COMMENT] |= ATT_CHG_PROPERTY_LOST;
+      comment_str = comment_node->info.value.data_value.str;
+      if (comment_str != NULL)
+	{
+	  attr_chg_prop.p[P_COMMENT] |= ATT_CHG_PROPERTY_DIFF;
+	  /* remove "LOST" flag */
+	  attr_chg_prop.p[P_COMMENT] &= ~ATT_CHG_PROPERTY_LOST;
+	}
+
+      if ((attr_chg_prop.name_space == ID_SHARED_ATTRIBUTE && attr_chg_prop.new_name_space == ID_ATTRIBUTE)
+	  || (attr_chg_prop.name_space == ID_ATTRIBUTE && attr_chg_prop.new_name_space == ID_SHARED_ATTRIBUTE))
+	{
+	  error = ER_ALTER_CHANGE_ATTR_TO_FROM_SHARED_NOT_ALLOWED;
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 1, attr_name);
+	  goto exit;
+	}
```

Comment:
is it necessary to handle this error in changing only column comment?

Reply:
When trying to change the comment of the ID_SHARED_ATTRIBUTE column, there was insufficient analysis to see if there was a problem.
I left the code of the build_attr_change_map function as it is for exception handling.
I think it's unnecessary because it doesn't handle shard configuration after version 10, but I'm not sure.

Reply:
it's like routine that checks that column properties change different.  I thought it would be unnecessary because only the comment is changed here.
Can name_space and new_name_space have different values?
Well... if it's not analyzed correctly, I don't think you need to remove the check routine

Reply:
Do not change shard settings in PT_CHANGE_COLUMN_COMMENT. I thought it was unnecessary and I removed this code.